images/router/clear-route-status.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# This allows for the clearing of route statuses, routers don't clear the routes status so some may be stale.
+# Upon deletion of the routes status active routers will immediatly update with a vaild status
+
+clear_status() {
+    MY_JSON_BLOB=$(curl -s -X GET http://localhost:8001/oapi/v1/namespaces/$1/routes/$2/)
+    if [[ $(echo ${MY_JSON_BLOB} |jq '.kind') == '"Route"' ]]; then
+        MODIFIED_JSON=$(echo "$MY_JSON_BLOB" | jq 'del(.status.ingress)')
+        curl -s -X PUT http://localhost:8001/oapi/v1/namespaces/$1/routes/$2/status --data-binary "${MODIFIED_JSON}" -H "Content-Type: application/json" > /dev/null
+        echo "route status for route $2 in namespace $1 cleared"
+    else
+        echo "Route $2 not found in namespace $1"
+    fi
+}
+
+cleanup() {
+    kill $PROXY_PID
+}
+trap cleanup EXIT
+
+USAGE="Usage:
+./clear-route-status.sh [namespace] [route-name | ALL]
+
+To clear the status of all routes in all namespaces:
+oc get namespaces | awk '{if (NR!=1) print \$1}' | xargs -n 1 -I %% ./clear-route-status.sh %% ALL
+
+To clear the status of all routes in namespace default:
+./clear-route-status.sh defalt ALL
+
+To clear the status of route example in namespace default:
+./clear-route-status.sh default example
+
+NOTE: if a router that admits a route is running it will immediatly update the cleared route status 
+"
+
+
+if [[ ${#} -ne 2 ]] || [[ ${@} == *" help "* ]]; then
+    printf "$USAGE"
+    exit
+fi
+
```

Comment:
Somewhere here declare your inputs:

```sh
namespace="${1}"
route_name="${2:-}"
```

So you're not using `$1` and `$2` in the rest of the script, which helps for readability.